packages/assets-controllers/src/TokensController.ts

```diff
@@ -115,6 +121,33 @@ export interface TokensState extends BaseState {
   suggestedAssets: SuggestedAssetMeta[];
 }
 
+/**
+ * The name of the {@link TokensController}.
+ */
+const controllerName = 'TokensController';
+
+/**
+ * The external actions available to the {@link TokensController}.
+ */
+type AllowedActions =
+  | AddApprovalRequest
+  | AcceptApprovalRequest
+  | RejectApprovalRequest;
+
+/**
+ * The messenger of the {@link TokensController}.
+ */
+export type TokensControllerMessenger = RestrictedControllerMessenger<
+  typeof controllerName,
+  AllowedActions,
+  never,
+  AllowedActions['type'],
+  never
+>;
+
+const ORIGIN_METAMASK = 'metamask';
```

Comment:
This is going to be a popular one, is it worth moving it to `controller-utils` in `constants.ts`?